core/bot_service.go

```diff
@@ -0,0 +1,236 @@
+package core
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"path"
+
+	"github.com/mr-tron/base58/base58"
+	shared "github.com/textileio/go-textile-core/bots"
+	"github.com/textileio/go-textile/crypto"
+	ipfs "github.com/textileio/go-textile/ipfs"
+)
+
+// BotIpfsHandler is the ipfs interface shared with a bot. It requires Get and Add
+type BotIpfsHandler struct {
+	botID string
+	node  *Textile
+}
+
+// BotKVStore is the kv store interface shared with a bot. Set, Get, Delete required
+type BotKVStore struct {
+	botID      string
+	botVersion int
+	node       *Textile
+}
+
+// Get allows a bot to get IPFS data by the cid/path. Allows optional key for decryption on the fly
+func (mip BotIpfsHandler) Get(pth string, key string) ([]byte, error) {
```

Comment:
why not a pointer method?

Reply:
well... short answer, i don't know. worth exploring more but i banged my head against the wall for a bit when first figuring that out. the plugin does not like it when i use pointers for these shared methods